crates/interpreter/src/valid.rs

```diff
@@ -503,22 +507,53 @@ impl<'a, 'm> Expr<'a, 'm> {
         match instr {
             Unreachable => self.stack_polymorphic(),
             Nop => (),
-            Block(b) => self.push_label(self.blocktype(&b)?, LabelKind::Block)?,
-            Loop(b) => self.push_label(self.blocktype(&b)?, LabelKind::Loop)?,
+            Block(b) => self.push_label(self.blocktype(&b)?, LabelKind::Block, None, false)?,
+            Loop(b) => self.push_label(self.blocktype(&b)?, LabelKind::Loop, None, false)?,
             If(b) => {
                 self.pop_check(ValType::I32)?;
-                self.push_label(self.blocktype(&b)?, LabelKind::If)?;
+                self.push_label(
+                    self.blocktype(&b)?,
+                    LabelKind::If,
+                    Some(self.side_table.len()),
+                    false,
+                )?;
+                self.side_table.push(SideTableEntryView {
+                    delta_ip: -self.instr_idx,
+                    // TODO(dev/fast-interp): implement below.
+                    delta_stp: 0,
+                    val_cnt: 0,
+                    pop_cnt: 0,
+                });
             }
             Else => {
+                let side_table_idx = self.immutable_label().side_table_idx.unwrap();
```

Comment:
I'm not sure why you need `immutable_label()`. You just get something that is not borrowed out of it. Using `label()` should just work.